src/providers/saml-jackson.ts

```diff
@@ -0,0 +1,51 @@
+import type { OAuthConfig, OAuthUserConfig } from "."
+
+export interface SAMLJacksonProfile {
+  id: string
+  email: string
+  firstName: string
+  lastName: string
+  name: string
+  email_verified: boolean
+  image: null
+}
+
+export default function SAMLJackson<
+  P extends Record<string, any> = SAMLJacksonProfile
+>(options: OAuthUserConfig<P>): OAuthConfig<P> {
+  return {
+    id: "saml-jackson",
+    name: "BoxyHQ SAML Jackson",
+    type: "oauth",
+    version: "2.0",
+    checks: ["pkce", "state"],
+    authorization: {
+      url: `${options.issuer}/api/oauth/authorize`,
+      params: {
+        scope: "",
+        response_type: "code",
+        provider: "saml",
+      },
+    },
+    token: {
+      url: `${options.issuer}/api/oauth/token`,
+      params: { grant_type: "authorization_code" },
+    },
+    userinfo: `${options.issuer}/api/oauth/userinfo`,
+    profile(profile) {
+      return {
+        id: profile.id || "",
+        firstName: profile.firstName || "",
+        lastName: profile.lastName || "",
+        email: profile.email || "",
+        name: `${profile.firstName || ""} ${profile.lastName || ""}`.trim(),
+        email_verified: true,
+        image: null,
```

Comment:
```suggestion
        id: profile.id,
        email: profile.email ?? null,
        name: [profile.firstName, profile.lastname].filter(Boolean).join(" "),
        image: null,
```

The initial profile is the same for all providers on purpose.

What's the reasoning for `|| ""` on multiple parameters? I would assume at least an `id` is always returned. :thinking: Is there a chance of that not being returned? That would complicate flows where the user wants to use a [database adapter](https://next-auth.js.org/adapters/overview).

What about email? Is it also not guaranteed to return?

Reply:
id and email are guaranteed to return.